src/plugins/knex.js

```diff
@@ -2,22 +2,144 @@
+          span.finish()
+        }
+      })
+    }
+  }
+}
+
+function createWrapSchemaBuilderToSQL (tracer, config) {
+  return function wrapToSQL (toSQL) {
+    return function toSQLWithTrace () {
+      const scope = tracer.scope()
+      const childOf = scope.active()
+
+      const span = tracer.startSpan('knex.SchemaBuilder.toSQL', {
+        childOf,
+        tags: {
+          'component': 'knex'
+        }
+      })
+      setDBTags(this, span)
+      setMethods(this, span)
+
+      return scope.activate(span, () => {
+        try {
+          return toSQL.apply(this, arguments)
+        } catch (e) {
+          throw addError(span, e)
+        } finally {
+          span.finish()
+        }
+      })
+    }
+  }
+}
+
+function addError (span, error) {
+  if (error) {
+    span.setTag('error', 'true')
+    span.log({
+      'error.type': error.name,
+      'error.msg': error.message,
+      'error.stack': error.stack
+    })
+  }
+  return error
+}
+
+function setDBTags (obj, span) {
+  if (obj.client && obj.client.config) {
+    const config = obj.client.config
+    if (config.client) {
+      span.setTag('db.type', config.client)
+    }
+    if (config.connection && config.connection.user) {
+      span.setTag('db.user', config.connection.user)
+    }
+  }
+}
+
+function setMethods (obj, span) {
+  if (obj._sequence) {
+    const methods = []
+    for (let i = 0, l = obj._sequence.length; i < l; i++) {
+      methods.push(obj._sequence[i].method)
+    }
+    span.setTag('schema.methods', methods)
+  }
+}
+
+function patchQueryToSQL (file) {
   return {
     name: 'knex',
     versions: ['>=0.8.0'],
     file,
     patch (Builder, tracer, config) {
       this.wrap(Builder.prototype, 'then', tx.createWrapThen(tracer, config))
+      this.wrap(Builder.prototype, 'toSQL', createWrapQueryBuilderToSQL(tracer, config))
     },
     unpatch (Builder) {
       this.unwrap(Builder.prototype, 'then')
+      this.unwrap(Builder.prototype, 'toSQL')
+    }
+  }
+}
+
+function patchSchemaToSQL (file) {
+  return {
+    name: 'knex',
+    versions: ['>=0.8.0'],
+    file,
+    patch (Builder, tracer, config) {
+      this.wrap(Builder.prototype, 'then', tx.createWrapThen(tracer, config))
+      this.wrap(Builder.prototype, 'toSQL', createWrapSchemaBuilderToSQL(tracer, config))
+    },
+    unpatch (Builder) {
+      this.unwrap(Builder.prototype, 'then')
+      this.unwrap(Builder.prototype, 'toSQL')
+    }
+  }
+}
+
+function patchRaw (file) {
+  return {
+    name: 'knex',
+    versions: ['>=0.8.0'],
+    file,
+    patch (Raw, tracer, config) {
+      this.wrap(Raw.prototype, 'then', tx.createWrapThen(tracer, config))
+    },
+    unpatch (Raw) {
+      this.unwrap(Raw.prototype, 'then')
     }
   }
 }
 
 module.exports = [
-  createPatch('lib/query/builder.js'),
-  createPatch('lib/raw.js'),
-  createPatch('lib/schema/builder.js')
+  patchQueryToSQL('lib/query/builder.js'),
+  patchSchemaToSQL('lib/schema/builder.js'),
+  patchRaw('lib/raw.js')
```

Comment:
Since these are now using different wrappers, using the standard pattern of an array of instrumentation objects makes more sense to me than the what's returned by the `patch___` functions.